src/routes/news/components/_tests_/NewsList.spec.js

```diff
@@ -0,0 +1,201 @@
+  }
+
+  it('should display error if the route has the status error', () => {
+    const state: StateType = prepareState({
+      status: 'error',
+      language: language.code,
+      city: city.code,
+      path: null,
+      type: 'local',
+      message: 'Something went wrong with the route',
+      code: ErrorCodes.UnknownError
+    })
+    expectError(state, ErrorCodes.UnknownError)
+  })
+
+  it('should display error if cities could not be loaded', () => {
+    const state: StateType = prepareState(successfulRouteState, {
+      cities: {
+        status: 'error',
+        message: 'Something went wrong with the cities',
+        code: ErrorCodes.UnknownError
+      }
+    })
+    expectError(state, ErrorCodes.UnknownError)
+  })
+
+  const expectLoadingIndicator = (state: StateType) => {
+    const navigation = createNavigationScreenPropMock()
+    navigation.state.key = 'route-id-0'
+    const store = mockStore(state)
+    jest.doMock('../NewsList', () => MockNewsList)
+    const NewsContainer = require('../../containers/NewsContainer').default
+    const result = TestRenderer.create(
+      <Provider store={store}><NewsContainer navigation={navigation} /></Provider>
+    )
+    jest.advanceTimersByTime(LOADING_TIMEOUT)
+    const indicator = result.root.findByType(ActivityIndicator)
+    expect(indicator).toBeTruthy()
+  }
+
+  it('should display loading indicator if the route is loading long enough', () => {
+    const state: StateType = prepareState({
+      path: null,
+      status: 'loading',
+      type: 'local',
+      language,
+      city
+    })
+    expectLoadingIndicator(state)
+  })
+
+  it('should display loading indicator if switching languages lasts long enough', () => {
+    const state: StateType = prepareState(successfulRouteState, { switchingLanguage: true })
+    expectLoadingIndicator(state)
+  })
+
+  it('should display loading indicator if cities are loading long enough', () => {
+    const state: StateType = prepareState(successfulRouteState, { cities: { status: 'loading' } })
+    expectLoadingIndicator(state)
+  })
+
+  it('should display loading indicator if languages are loading long enough', () => {
+    const state: StateType = prepareState(successfulRouteState, { languages: { status: 'loading' } })
+    expectLoadingIndicator(state)
+  })
+
+  it('should display NewsListItem component if the state is ready', () => {
+    const state: StateType = prepareState(successfulRouteState)
+    const store = mockStore(state)
+    const navigation = createNavigationScreenPropMock()
+    navigation.state.key = 'route-id-0'
+    jest.doMock('../NewsList', () => MockNewsList)
+    const NewsContainer = require('../../containers/NewsContainer').default
+    const result = TestRenderer.create(
+      <Provider store={store}><NewsContainer navigation={navigation} /></Provider>
+    )
+    const newsListInstance = result.root.findByProps({ newsList: newsList })
+
+    expect(newsListInstance).toBeTruthy()
+  })
+})
```

Comment:
I think all of these tests are rather tests of the `NewsContainer` and not the `NewsList`.

Reply:
Yes, imo that would be right

Reply:
Done